dra-evolution/pkg/api/capacity_types.go

```diff
@@ -64,12 +64,33 @@ type ResourcePoolSpec struct {
 	// Must not have more than 128 entries.
 	Devices []Device `json:"devices,omitempty"`
 
+	// SharedCapacity are pooled resources that are shared by all
+	// devices in the pool. This is typically used when representing a
+	// partitionable device, and need not be populated otherwise.
+	//
+	// Must not have more than 32 entries.
+	//
+	// +optional
+	// +listType=atomic
+	SharedCapacity []SharedCapacity `json:"sharedCapacity,omitempty"`
+
 	// FUTURE EXTENSION: some other kind of list, should we ever need it.
 	// Old clients seeing an empty Devices field can safely ignore the (to
 	// them) empty pool.
 }
 
+type ResourceCapacity struct {
+	// Name is the resource name/type.
+	// +required
+	Name string `json:"name"`
+
+	// Capacity is the total capacity of the named resource.
+	// +required
+	Capacity resource.Quantity `json:"capacity"`
```

Comment:
I prefer not using a `resource.Quantity`. As pointed out elsewhere, this is not a user-facing API. It's for the scheduler machinery. A simpler `Count int64` would be easier to encode and work with.

If there ever is a need for a value < 1, then the driver can scale both capacity and consumed values up by a fixed factor (1000, 1000000).